signalfx-java/src/main/java/com/signalfx/signalflow/ComputationHandler.java

```diff
@@ -0,0 +1,147 @@
+package com.signalfx.signalflow;
+
+import java.util.concurrent.Callable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.signalfx.signalflow.ChannelMessage.DataMessage;
+import com.signalfx.signalflow.ChannelMessage.EventMessage;
+import com.signalfx.signalflow.ChannelMessage.JobProgressMessage;
+import com.signalfx.signalflow.ChannelMessage.JobStartMessage;
+import com.signalfx.signalflow.ChannelMessage.MetadataMessage;
+import com.signalfx.signalflow.Computation.State;
+
+/**
+ * Copyright (C) 2016 SignalFx, Inc. All rights reserved.
+ * 
+ * Class that provides basic plumbing used by subclasses to processing computation
```

Comment:
how so?  since it is an abstract class the subclass will implement any of the onmessage methods they want.  or did you mean the class comment was unfinished?